cmd/compute-domain-controller/computedomain.go

```diff
@@ -215,9 +219,38 @@ func (m *ComputeDomainManager) AssertWorkloadsCompleted(ctx context.Context, cdU
 	}
 
 	if len(nodes.Items) != 0 {
+		// show nodes with labels
+		nodeNames := []string{}
+		for _, node := range nodes.Items {
+			nodeNames = append(nodeNames, node.Name)
+		}
+		klog.Errorf("Found %d nodes with label for ComputeDomain with UID  %s: %v", len(nodes.Items), cdUID, nodeNames)
 		return fmt.Errorf("nodes exist with label for ComputeDomain %s", cdUID)
```

Comment:
Do not add logs next to lines that return an error -- the errors get logged at the top of the workqueue. Also, if you want to include the list of node names in the error message, please create a helper to gather the node names, rather than introducing extra code into this if statement block.

Reply:
Yes yes i see this as redundant. I am going to clean it (playing a bit). Infact i should add the extra info in the error message. 

Noted about the helper fun